acceptance/builder_test.go

```diff
@@ -0,0 +1,367 @@
+			it("will replace placeholder with default planPath", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_PLAN_PATH=<layers>/plan.toml",
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				h.AssertNotNil(t, err)
+				expected := "failed to parse detect plan: open /layers/plan.toml"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+	})
+
+	/// .../cmd/lifecycle/builder.go#build
+	when("Builder args are successfully transmitted to in build script", func() {
+		when("CNB_APP_DIR changed", func() {
+			it("CNB_APP_DIR is successfully transmitted to in build script", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					"--env", "CNB_APP_DIR=/env_folders/different_cnb_app_dir_from_env",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err) //due to not exist directory
+				expected := "CNB_APP_DIR: /env_folders/different_cnb_app_dir_from_env"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		when("CNB_BUILDPACKS_DIR changed", func() {
+			it("CNB_BUILDPACKS_DIR is successfully transmitted to in build script", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					"--env", "CNB_BUILDPACKS_DIR=/env_folders/different_buildpack_dir_from_env",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err) //due to not exist directory
+				expected := "CNB_BUILDPACK_DIR: /env_folders/different_buildpack_dir_from_env"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		when("CNB_LAYERS_DIR", func() {
+			it("CNB_LAYERS_DIR is successfully transmitted to in build script", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					"--env", "CNB_LAYERS_DIR=/tmp/different_layers_path_dir_from_env",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err) //due to not exist directory
+				expected := "layers_dir: /tmp/different_layers_path_dir_from_env"
```

Comment:
The layers directory provided to an individual buildpack is actually going to be a sub directory of `CNB_LAYERS_DIR`.

Reply:
```suggestion
				expected := "layers_dir: /tmp/different_layers_path_dir_from_env/hello_world"
```

Reply:
fixed too